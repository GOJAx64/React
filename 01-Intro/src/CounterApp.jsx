import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ( {value} ) => {
    const [counter, setCounter] = useState(value);
    
    const addValue = () => setCounter(counter + 1);
    //We can do it different, try ->  setCounter( (c) => c + 1 )

    const subsValue = () => setCounter(counter - 1);

    const resetValue = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2> 
            <button onClick={ addValue }> +1 </button>  
            <button onClick={ subsValue }> -1 </button>
            <button onClick={ resetValue }> Reset </button>
        </>
  )
}


CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
    value: 0,
}