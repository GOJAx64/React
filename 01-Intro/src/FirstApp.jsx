import PropTypes from 'prop-types';

//We can use fragmet, but it's not used a lot
const message = {
    age: 22,
    me: 'Alberto', 
}

const printMessage = (message) => message.age + ' ' + message.me

export const FirstApp = ( {title, number, subtitle} ) => {
  return (
    <>
        <h1>{ title }</h1>
        {/*<h1> { printMessage(message) } </h1>
        <p>This is a subtitle</p>
        <br/>*/}
        <p>{ subtitle }</p>
        <p>{ number }</p>
    </>
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    number: PropTypes.number.isRequired,
}

FirstApp.defaultProps = {
    //title: 'Alberto Garc√≠a',
    number: 888,
    //subtitle: 'This is a default propertie',
}
