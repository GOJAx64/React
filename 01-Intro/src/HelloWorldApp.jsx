/*export function App() { //My first component
    return (<h1>Hello World</h1>)
}*/

//We can use this
//export default App;
//use rafc
export const HelloWorldApp = () =>  {
    return (
        <p>HelloWorld App</p>
    )
}
