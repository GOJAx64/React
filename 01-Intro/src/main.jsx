import React from 'react'
import ReactDOM from 'react-dom/client'
import {HelloWorldApp} from './HelloWorldApp'
import { FirstApp } from './FirstApp'
import './styles.css'
import { CounterApp } from './CounterApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/*<HelloWorldApp/>
    <FirstApp title='title props' number={ 123 }/> */}
    <CounterApp value={0}/>
  </React.StrictMode>
)
