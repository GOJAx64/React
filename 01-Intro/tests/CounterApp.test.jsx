import { render, screen, fireEvent } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Tests in <CounterApp/>', () => {  

    const initialValue = 0;

    test('should do match with the SnapShot', () => { 
        const { container } = render(<CounterApp value={ initialValue } />);
        expect(container).toMatchSnapshot();
    });

    test('should show the initial value as 100', () => { 
        render(<CounterApp value={ 100 } />);
        expect(screen.getByText(100)).toBeTruthy();
    });

    test('should increase with the +1 button', () => { 
        render(<CounterApp value={ initialValue} />);
        fireEvent.click( screen.getByText('+1') );
        expect( screen.getByText('1') ).toBeTruthy();
    });

    test('should decrease with the -1 button', () => {
        render( <CounterApp value={ initialValue } /> );
        fireEvent.click( screen.getByText('-1') );
        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain('-1');
        //expect( screen.getByText('-1') ).toBeTruthy();
    });

    test('should works the reset button', () => {
        render( <CounterApp value={ 355 } /> );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('Reset') );
        //fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
        expect( screen.getByText( 355 ) ).toBeTruthy();
    });
});