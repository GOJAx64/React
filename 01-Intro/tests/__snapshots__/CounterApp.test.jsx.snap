// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <CounterApp/> should do match with the SnapShot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    0
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button>
     Reset 
  </button>
</div>
`;
